contracts/Core/JobManager.sol

```diff
@@ -37,79 +38,209 @@ contract JobManager is ACL, JobStorage {
         Structs.Job memory job = Structs.Job(
-            numJobs,
-            epoch,
-            url,
-            selector,
-            name,
-            repeat,
-            msg.sender,
-            msg.value,
-            false,
-            0
+            numAssets, 
+            epoch, 
+            url, 
+            selector, 
+            name, 
+            repeat, 
+            msg.sender, 
+            msg.value, 
+            false, 
+            0,
+            uint256(assetTypes.Job)
         );
-        jobs[numJobs] = job;
-        
+        jobs[numAssets] = job;
+        jobList.push(numAssets);
+
         emit JobCreated(
-            numJobs,
-            epoch,
-            url,
-            selector,
-            name,
-            repeat,
-            msg.sender,
-            msg.value,
-            block.timestamp
+            numAssets, 
+            epoch, 
+            url, 
+            selector, 
+            name, 
+            repeat, 
+            msg.sender, 
+            msg.value, 
+            block.timestamp, 
+            uint256(assetTypes.Job)
         );
+        // jobs.push(job);
     }
 
-    function fulfillJob(
-        uint256 jobId,
-        uint256 value
-    )
+    function fulfillAsset(
+        uint256 id, 
+        uint256 value   
+    ) 
         external 
         onlyRole(Constants.getJobConfirmerHash())
     {
-        Structs.Job storage job = jobs[jobId];
         uint256 epoch = stateManager.getEpoch();
+        if(jobs[id].assetType==uint256(assetTypes.Job)){
+
+            Structs.Job storage job = jobs[id];
+
+            if (!job.repeat) {
+                job.fulfilled = true;
+            }
 
-        if (!job.repeat) {
-            job.fulfilled = true;
+            job.result = value;
+            emit JobReported(
+                job.id,
+                value,
+                epoch,
+                job.url,
+                job.selector,
+                job.name,
+                job.repeat,
+                job.creator,
+                job.credit,
+                job.fulfilled,
+                block.timestamp
+            );
         }
+        else if(collections[id].assetType==uint256(assetTypes.Collection)){
 
-        job.result = value;
-        emit JobReported(
-            job.id,
-            value,
+            Structs.Collection storage collection = collections[id];
+
+            collection.result = value;
+
+            emit CollectionReported(
+                collection.id,
+                value,
+                epoch,
+                collection.name,
+                collection.aggregationMethod,
+                collection.jobIDs,
+                collection.creator,
+                collection.credit,
+                block.timestamp
+            );
+        }
+    }
+
+    function createCollection(
+        string calldata name, 
+        uint256[] memory jobIDs,
+        uint32 aggregationMethod
```

Comment:
@SamAg19

Reply:
I dont think we can input an enum in the function and the fact that this will be stored as an integer in the struct, i don't think so it makes sense to use an enum here. Whats your idea of using an enum here?